plugins/trader/portfolioManager.js

```diff
@@ -17,381 +15,95 @@ var log = require(dirs.core + 'log');
-    }
-
-    log.info(this.action, 'was successfull');
-
-    this.relayOrder();
-  }
-
-  var handleCancelResult = function(alreadyFilled) {
-    if(alreadyFilled)
-      return;
-
-    if(this.exchangeMeta.forceReorderDelay) {
-        //We need to wait in case a canceled order has already reduced the amount
-        var wait = 10;
-        log.debug(`Waiting ${wait} seconds before starting a new trade on ${this.exchangeMeta.name}!`);
-
-        setTimeout(
-            () => this.trade(this.action, true),
-            +moment.duration(wait, 'seconds')
-        );
-        return;
-    }
-
-    this.trade(this.action, true);
-  }
-
-  this.exchange.checkOrder(_.last(this.orders), _.bind(handleCheckResult, this));
-}
-
-// convert into the portfolio expected by the performanceAnalyzer
-Manager.prototype.convertPortfolio = function(portfolio) {
-  var asset = _.find(portfolio, a => a.name === this.asset).amount;
-  var currency = _.find(portfolio, a => a.name === this.currency).amount;
-
-  return {
-    currency,
-    asset,
-    balance: currency + (asset * this.ticker.bid)
-  }
-}
-
-Manager.prototype.relayOrder = function(done) {
-  // look up all executed orders and relay average.
-  var relay = (err, res) => {
-
-    var price = 0;
-    var amount = 0;
-    var date = moment(0);
-
-    _.each(res.filter(o => !_.isUndefined(o) && o.amount), order => {
-      date = _.max([moment(order.date), date]);
-      price = ((price * amount) + (order.price * order.amount)) / (order.amount + amount);
-      amount += +order.amount;
-    });
-
-    async.series([
-      this.setPortfolio,
-      this.setTicker
-    ], () => {
-      const portfolio = this.convertPortfolio(this.portfolio);
-
-      this.emit('trade', {
-        date,
-        price,
-        portfolio: portfolio,
-        balance: portfolio.balance,
-
-        // NOTE: within the portfolioManager
-        // this is in uppercase, everywhere else
-        // (UI, performanceAnalyzer, etc. it is
-        // lowercase)
-        action: this.action.toLowerCase()
-      });
-
-      this.orders = [];
-
-      if(_.isFunction(done))
-        done();
-    });
+// instantiate a new trade object
+// TODO - impliment different trade execution types / strategies
+//        by invoking variable trade subclasses
+//      - pass the trade a specific amount limited by a currency allowance
+Manager.prototype.newTrade = function(what) {
+  log.debug("PORTFOLIO MANAGER : newTrade() : creating a new Trade class to ", what, this.conf.asset, "/", this.conf.currency)
 
+  // push the current (asummed to be inactive) trade to the history
+  if(this.currentTrade){
+    this.tradeHistory.push(this.currentTrade)
   }
 
-  var getOrders = _.map(
-    this.orders,
-    order => next => this.exchange.getOrder(order, next)
-  );
+  return this.currentTrade = new Trade({
+    action: what,
+    exchange:this.exchange,
+    currency: this.conf.currency,
+    asset: this.conf.asset,
+    portfolio: this.portfolio,
```

Comment:
I do not like all of these property values passed in, ideally it's just `action`, `currency`, `asset`, and then an `amount` which could have an undefined value which would then default to the system settings for currency allowance, keep asset, etc.

@askmike You know the architecture best, is there an currently established way by which we could store references to globally accessible objects / dependencies without resorting to a singleton?